### DISCUSSION TITLE
Add Creative Commons licenses as Off-Chain Data Standard

### DISCUSSION THREAD

**Discussion by jeemclr** [capsule-corp-ternoa/ternoa-proposals#43] [Ideas]
Creative Commons licences are a hot topic for NFT projects. We see many projects switch to CC-0 and put their project in the public domain.

However, it is difficult to find the terms and conditions of projects and they are never registered in the blockchain.

I suggest the ability to register the copyrights on-chain, by adding a hash allowing to link the copyrights to each NFT. Also, can be text value as licences. 

The six licenses (excluding CC-0 which is an equivalent to the Public Domain) are:

CC-BY
CC-BY-SA
CC-BY-ND
CC-BY-NC
CC-BY-NC-SA
CC-BY-NC-ND

The following chart illustrates the permissions allowed by each license.

![ccinfographic3withlicense](https://user-images.githubusercontent.com/6671716/185198980-66f7f454-934c-4e36-9b6b-4c21f33272aa.jpg)

**Comment by markopoloparadox** [comment]
Yeah, this is something @peshwar9  and I have discussed today. Solution that we have found is to have a separate pallet which is going to store links to licenses. Then NFTs would use indicies to represent what kind of license it has. 
Example
License Table
```
0. https://creativecommons.org/licenses/by/4.0/
1. https://creativecommons.org/licenses/by-sa/4.0/
2. https://ipfs.ternoa.dev/ipfs/QmVJFhp7APxaHtCoypoqUr3L9A5f88BDfJmoJDLuRnpV1M
3. my.aws.server/whatever
```

NFTData
```
NFTData {
  owner: XXYY
  creater: XXYY
  license: 1
}
```

Now there are some edge cases that we need to work out.
1) When the users creates an NFT does he need to specify a license?
2) Is there a default license?
3) What should we do with NFTs that were created before this feature was implemented? 
4) Once an NFT has a license can it be changed?
5) Who is going to be allowed to add new licenses to the table? If everyone will be able to do it is there going to be a License Mint fee?

**Reply by jeemclr** [reply]
To answer your questions here my thoughts:
1. No. It's suggested bur not required. What is the default legal agreement if nothing is set?
2. Think differ form countries. To be checked with a legal person
3. Default licensing
4. Yes. As we can see on some famous NFTs project (BAYC, Moonbirds, Cryptopunks, etc.) have changed their policy about creative commons for a more "web3" compliant so it can be important that those informations could be **mutable** 
5. Only creator I guess

**Reply by Leouarz** [reply]
Are License added by us or users ?

**Reply by Leouarz** [reply]
Yes !

**Reply by peshwar9** [reply]
@M2613 What is your view on legal jurisdictions associated with NFT licenses?

**Comment by markopoloparadox** [comment]
@M2613 If I will have some free time I will draft a initial spec version. It shouldn't be too hard.

@Leouarz This feature could be your task

**Reply by Leouarz** [reply]
Yeah, why not, should not be too hard after drafting some specs.

**Comment by Leouarz** [comment]
I think we should link licenses to collections : 

```rust
pub struct LicenseData {
  offchainData: BoundedVec<u8, LicenseOffchainDataLimit>,
  creator: AccountId
}

pub type Licenses: StorageMap<LicenseId, LicenseData>;

pub type CollectionLicenses: StorageMap<CollectionId, LicenseId>;
```

What do you think @peshwar9 @markopoloparadox

**Reply by peshwar9** [reply]
@Leouarz Yeah, it makes sense to link licences to collections in general, as they can apply to PfP collections to start with. But could it be useful for individual NFTs too? e.g.  a licensing agreement associated with a music single or a secret NFT?

**Reply by Leouarz** [reply]
Thinking about it, for a single NFT, it could be linked to a collection (since it has no fees).
Now the question is more : Can 2 NFTs in the same collection have different licenses

**Comment by peshwar9** [comment]
No that can make things complicated, and I don’t see whichbuse case might
need this. If there are different licenses, maybe they should be in
different collections?

**Reply by markopoloparadox** [reply]
I agree. A collection is something that is quite flexible and you could have two collections having similar NFTs but different licenses.
